packages/sitecore-jss-react/src/components/sharedTypes.ts

```diff
@@ -4,7 +4,10 @@ import { ComponentType } from 'react';
  * @param {string} componentName component to be imported from the component factory
  * @param {string?} exportName component to be imported in case you export multiple components from the same file
  */
-export type ComponentFactory = (componentName: string, exportName?: string) => ComponentType | null;
+export type ComponentFactory = (
+  componentName: string,
+  exportName?: string
+) => JssComponentType | null;
 
 /**
  * Component type returned from component builder / factory
```

Comment:
_render_ property for dynamic components, is it a next.js component specific property or it's also present when using react dynamic components?

Reply:
_preload_ property is used but not defined in a type

Reply:
> is render a next.js component specific property
to my knowledge, yes. That's why it is defined as optional and the sub-properties are not explicitly defined, using the catch-all type of `{ [key: string]: unknown }` instead